app/controllers/users_controller.rb

```diff
@@ -0,0 +1,48 @@
+#
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def index
+    @users = User.all
+  end
+
+  def show
+    @user = User.find(params[:id])
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      session[:user_id] = @user.id
+      flash[:success] = 'Signed Up'
+      redirect_to @user
+    else
+      render 'static_pages/error'
```

Comment:
We'd want to re-render the user create form to let the user fix errors... not forward the user to an errors page.